### ISSUE TITLE
Logstash (1.4.2) ES (1.4.2) Connection issues

### ISSUE THREAD

**Issue by krinkere** [elastic/logstash#2617]
Related to this one: https://github.com/elasticsearch/logstash/issues/1821
but i dont think it is documentation issue

ES 1.4.2
Logstash 1.4.2
According to http://logstash.net/docs/1.4.2/outputs/elasticsearch
VERSION NOTE: Your Elasticsearch cluster must be running Elasticsearch 1.0.0 or later

I should be fine... but I ran it on Centos 6.5 and same thing. When I try

/opt/logstash/bin/logstash -e 'input {stdin{}} output{elasticsearch{host=>localhost}}'

I get this:

log4j, [2015-02-15T05:41:51.291]  WARN: org.elasticsearch.discovery: [logstash-c6501.ambari.apache.org-22688-2010] waited for 30s and no initial state was set by the discovery

Basically suggesting that I have connection issues.

I tried to flip between multicast and unicast. Same thing
Set up for multicast - default /etc/logstash/conf.d/logstash.conf used

Then I set
discovery.zen.ping.multicast.enabled: false
discovery.zen.ping.unicast.hosts: ["192.168.65.101"]

This one works:
/opt/logstash/bin/logstash -e 'input {stdin{}} output{elasticsearch{host=>"192.168.65.101" protocol=>"http" port=>"9200"}}'

or if I use elasticsearch_http (which is the same thing)

**Comment by untergeek** [comment]
Unless there are firewall rules in place blocking ports 9300-9400, Logstash 1.4.2 should connect to _any_ Elasticsearch cluster of v1.x.x via the node protocol. 

We actually don't recommend  using multicast in a production environment. I do recommend using `protocol => "http"`.

The node protocol (the default behavior) spins up a local client node which tries to join the elasticsearch cluster using multicast (also the default), unless you specify a host. If your Elasticsearch cluster is named something other than `elasticsearch` you need to add 'cluster => "myclustername"` to your config. We _highly_ recommend not having your cluster name be left unconfigured.

**Comment by krinkere** [comment]
Thanks for the reply.
I am pretty sure that my firewalls are not blocking 9300-9400, but I can double check that.

Yeah, definitely no multicast or using default cluster, or node.name in production!

Ummm, I am not in front of my system, but I feel that the issue is that I did rename my cluster, but I am not sure if I specified cluster in my settings. Also something to double check.

**Comment by arei** [comment]
I was having a very similar issue to this earlier this evening getting the same error message as @krinkere describes.  After some initial testing it looks like logstash cannot find the ES cluster unless it is specifically pointed at a data node. Consider the following environment:

Server AAA: RHEL 6.5, ES, HTTP Only, No Data
Server BBB: RHEL 6.5, ES, Data Only, No HTTP
Server CCC: RHEL 6.5, Logstash

If I configure Logstash thus:

    output { 
      elasticsearch { 
        protocol => "node"
        cluster => "somecluster"
      }
    }

No luck.

If I add a host pointing at the http node:

    output { 
      elasticsearch { 
        protocol => "node"
        cluster => "somecluster"
        host => "AAA"
      }
    }

Still no dice.

Finally, if I point host at the data node:

    output { 
      elasticsearch { 
        protocol => "node"
        cluster => "somecluster"
        host => "BBB"
      }
    }

It connects and all is right with the world.

**Comment by krinkere** [comment]
Finally was able to test suggestions made by @untergeek 

So yeah, when i added cluster name (which was different from default "elasticsearch"), it worked like a charm. Interesting enough, it would not care about cluster name when specifying elasticsearch_http or using protocol.

But yeah, this worked:
/opt/logstash/bin/logstash -e 'input {stdin{}} output{elasticsearch{host=>localhost cluster=>"logstash"}}'

**Comment by torrancew** [comment]
@krinkere this is because in HTTP mode, you are addressing a member of a specific cluster directly (to overlook things like tribe nodes and such), so there is no need to specify any type of cluster name. When using the "node" protocol, you are actually having Logstash create an Elasticsearch instance that joins the cluster as a client member. Since cluster members must agree on the name of the cluster they belong to, this is where the requirement for that setting comes in.

**Comment by krinkere** [comment]
@torrancew - makes perfect sense. thank you!
